File: commands/remove.js
import { EmbedBuilder } from 'discord.js';
import db from '../services/db';
import colors from '../models/colors';
import log from '../utils/log';

export const helpers = {
	ephemeral: false,
};

export async function run(interaction) {
	if (interaction.guildId !== '669867414409969664') return;

	const id = interaction.options.getString('id');
	if (!id) return interaction.editReply({ content: 'ID не валидный!' });

	const guild = await bot.guilds.fetch(id);
	if (!guild) return interaction.editReply({ content: 'Гильдия не найдена!' });

	await guild.leave();
	await db.query('UPDATE guilds SET deleted_at = now() WHERE id = ?', [id]);

	log(`Removing guild: ${guild.name} (${guild.id})`);

	const embed = new EmbedBuilder()
		.setAuthor({
			name: 'Успешно! Сервер «${guild.name}» удален!',
			iconURL: guild.iconURL(),
		})
		.setColor(colors.green)
		.setTimestamp()
		.setFooter({ text: interaction.user.tag });

	await interaction.editReply({ embeds: [embed] });
}

export default {
	helpers,
	run,
};

File: commands/remove.go
package commands

import (
	"context"
	"fmt"
	"time"

	"github.com/bwmarrin/discordgo"

	"sdc/domain"
	"sdc/models"
	"sdc/utils"
)

func Remove(d *Deps, s *discordgo.Session, i *discordgo.InteractionCreate) error {
	if i.GuildID != domain.StaffGuildID {
		return nil
	}
	ctx := context.Background()
	id, ok := optionString(i, "id")
	if !ok || id == "" {
		return editReply(s, i, &discordgo.WebhookEdit{Content: strPtr("ID не валидный!")})
	}

	guild, err := s.Guild(id)
	if err != nil || guild == nil {
		return editReply(s, i, &discordgo.WebhookEdit{Content: strPtr("Гильдия не найдена!")})
	}

	if err := s.GuildLeave(id); err != nil {
		return err
	}
	if err := d.Store.SoftDeleteGuild(ctx, id); err != nil {
		return err
	}
	d.Cache.Delete(id)

	utils.Log("Removing guild: %s (%s)", guild.Name, guild.ID)

	author := &discordgo.MessageEmbedAuthor{
		Name: fmt.Sprintf("Успешно! Сервер «%s» удален!", guild.Name),
	}
	if guild.Icon != "" {
		author.IconURL = discordgo.EndpointGuildIcon(guild.ID, guild.Icon)
	}

	embed := &discordgo.MessageEmbed{
		Author:    author,
		Color:     models.EmbedColors["green"],
		Timestamp: time.Now().Format(time.RFC3339),
		Footer:    &discordgo.MessageEmbedFooter{Text: i.Member.User.String()},
	}
	return editReply(s, i, &discordgo.WebhookEdit{Embeds: &[]*discordgo.MessageEmbed{embed}})
}
